lib/pure/options.nim

```diff
@@ -108,6 +108,41 @@ proc get*[T](self: Option[T]): T =
     raise UnpackError(msg : "Can't obtain a value from a `none`")
   self.val
 
+proc get*[T](self: Option[T], otherwise: T): T =
+  ## Returns the contents of this option or `otherwise` if the option is none.
+  if self.isSome:
+    self.val
+  else:
+    otherwise
+
+
+proc each*[T](self: Option[T], callback: proc (input: T)) =
```

Comment:
I was originally against using `map` to perform an `each`, as I see them as separate functionalities. However, it looks like the precedent has already been set to have `map` do what an `each` does:

https://github.com/nim-lang/Nim/blob/devel/lib/system.nim#L2124-L2137

I'm not going to lie, this makes me itch. A `map` is an operation that produces a result, while an `each` operates without producing result. However, I think consistency within the core library trumps most things, so I'm willing to concede on this.